### PR TITLE
Fix Errors when sending to server

### PR DESCRIPTION
**Pull request by saramissak**

### CODE REVIEW

File: src/main/webapp/js/playGame.js [hunk 1]
```diff
@@ -23,8 +17,24 @@ async function initMapToPlayGame() {
...
+      stageID = data.stages[0];
+      initStage =  await getStage(stageID);
+      startingLocation = {lat: initStage.startingLocation.latitude, lng: initStage.startingLocation.longitude};
+    } else {
+      console.log('from progress');
+      stageID = userProgress.stageID;
+      initStage =  await getStage(stageID);
+      startingLocation = {lat: userProgress.location.latitude, lng: userProgress.location.latitude.longitude};
+      console.log(startingLocation);
+
```
Comment (ldcWV) [inline src/main/webapp/js/playGame.js]: delete empty line

File: src/main/webapp/js/playGame.js [hunk 2]
```diff
@@ -23,8 +17,24 @@ async function initMapToPlayGame() {
...
-    var initStage =  await getStage(data.stages[0]);
-    var startingLocation = {lat: initStage.startingLocation.latitude, lng: initStage.startingLocation.longitude};
+    var userProgress = getUserProgress();
+    var startingLocation;
+    var initStage;
+    var stageID;
+    console.log(userProgress);
+    if (userProgress == null) {
+      stageID = data.stages[0];
+      initStage =  await getStage(stageID);
```
Comment (ldcWV) [inline src/main/webapp/js/playGame.js]: Delete extra space.

File: src/main/webapp/js/playGame.js [hunk 3]
```diff
@@ -23,8 +17,24 @@ async function initMapToPlayGame() {
...
+      startingLocation = {lat: initStage.startingLocation.latitude, lng: initStage.startingLocation.longitude};
+    } else {
+      console.log('from progress');
+      stageID = userProgress.stageID;
+      initStage =  await getStage(stageID);
```
Comment (ldcWV) [inline src/main/webapp/js/playGame.js]: Delete extra space.

File: src/main/webapp/js/playGame.js [hunk 4]
```diff
@@ -52,11 +61,42 @@ async function initMapToPlayGame() {
...
+  });
+  return result;
+}
+
+/**
+* Gets the gameID from the URL
+* @return gameID from the URL or returns null and relocates you to index.html page
+*/
+function getGameID() {
+  const params = new URLSearchParams();
```
Comment (ldcWV) [inline src/main/webapp/js/playGame.js]: Unused variable?

**Review by ldcWV — APPROVED** [review]

File: src/main/webapp/js/playGame.js (hunk 2 above)
Comment (saramissak) [reply src/main/webapp/js/playGame.js]: Done. Thank you!

File: src/main/webapp/js/playGame.js (hunk 1 above)
Comment (saramissak) [reply src/main/webapp/js/playGame.js]: That is already removed in my current version, but yes thank you.

File: src/main/webapp/js/playGame.js (hunk 4 above)
Comment (saramissak) [reply src/main/webapp/js/playGame.js]: You're right, took that out!

**Pull request merged**